src/util/srp.cpp

```diff
@@ -429,7 +429,7 @@ static SRP_Result H_ns(mpz_t result, SRP_HashAlgorithm alg, const unsigned char
 	unsigned char *bin = (unsigned char *)srp_alloc(nbytes);
 	if (!bin) return SRP_ERR;
 	memcpy(bin, n, len_n);
-	memcpy(bin + len_n, bytes, len_bytes);
+	memcpy(bin + len_n, bytes, SHA512_DIGEST_LENGTH);
```

Comment:
even though it always is `== SHA512_DIGEST_LENGTH` pratice, you are silently truncating `len_bytes` here